File: scala/2348_Number_of_Zero_Filled_Subarrays.scala
object Solution {
    def zeroFilledSubarray(nums: Array[Int]): Long = {
        var total: Long = 0;
        var current: Long = 0;
        for (i <- 0 to (nums.length - 1)) {
            if (nums(i) == 0) {
                current += 1;
                total += current;
            } else {
                current = 0;
            }
        }
        return total;
    }
}

File: rust/2348_Number_of_Zero_Filled_Subarrays.rs
impl Solution {
    pub fn zero_filled_subarray(nums: Vec<i32>) -> i64 {
        let mut current: i64 = 0;
        let mut total: i64 = 0;
        for i in 0..nums.len() as usize {
            if (nums[i] == 0) {
                current += 1;
                total += current;
            } else {
                current = 0;
            }
        }  
        return total;
    }
}
